frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue

```diff
@@ -290,6 +301,20 @@ export default {
       this.showFullReactionListMissing = false;
       this.showFullReactionListMap = false;
     },
+    toggleSelectionCardContent() {
+      if (this.showSelectionCardContent) {
+        this.hideSelectionCardContent();
+      } else {
+        this.openSelectionCardContent();
+      }
+    },
+    openSelectionCardContent() {
+      this.showSelectionCardContent = true;
+      this.$emit('openSelectionCardContent');
```

Comment:
If there is no other way to implement the same behaviour without the `$emit`, then it would be acceptable to delete it and the subsequent scroll reset, and increase the timeout from 300ms to ~800ms.

Reply:
What is wrong with `$emit`?

Reply:
Emits have effects in a non-deterministic way, ie one cannot deduce where event is consumed. For this reason, they should be avoided.

Reply:
[`.sync`](https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier) is a deterministic `$emit` substitute - it could be used with a `isCollapsed` prop.

Reply:
Oh right! I totally confused the `$emit`s.
But now that I looked into it - do you know if the strict child-parent communication is enforced? I couldn't find  anything about the strictness in the documentation, but I did find ways around it eg https://stackoverflow.com/questions/42615445/vuejs-2-0-emit-event-from-grand-child-to-his-grand-parent-component .

Reply:
It's enforced in the sense that if you didn't have the binding `@openSelectionCardContent="resetSidebarLayout"` then it would not work since there are no listeners matching the emit.

There is a section about it here: https://vuejs.org/v2/guide/components.html#Listening-to-Child-Components-Events